pkg/system/init_windows.go

```diff
@@ -1,24 +1,20 @@
 package system // import "github.com/docker/docker/pkg/system"
 
-import (
-	"os"
-
-	"github.com/sirupsen/logrus"
-)
+import "github.com/Microsoft/hcsshim/osversion"
 
 var (
 	// containerdRuntimeSupported determines if ContainerD should be the runtime.
-	// As of March 2019, this is an experimental feature.
 	containerdRuntimeSupported = false
 )
 
-// InitContainerdRuntime sets whether to use ContainerD for runtime
-// on Windows. This is an experimental feature still in development, and
-// also requires an environment variable to be set (so as not to turn the
-// feature on from simply experimental which would also mean LCOW.
-func InitContainerdRuntime(experimental bool, cdPath string) {
-	if experimental && len(cdPath) > 0 && len(os.Getenv("DOCKER_WINDOWS_CONTAINERD_RUNTIME")) > 0 {
-		logrus.Warnf("Using ContainerD runtime. This feature is experimental")
+// InitContainerdRuntime sets whether to use ContainerD for runtime on Windows.
+func InitContainerdRuntime(cdPath string) {
+	if len(cdPath) > 0 {
+		containerdRuntimeSupported = true
+	}
+
+	// Always use containerd runtime on Windows Server 2022 preview and later
+	if len(cdPath) == 0 && osversion.Build() >= 20295 {
```

Comment:
This could be simply
```
if len(cdPath) > 0 || osversion.Build() >= 20295
```
which seems easier to scan for the intended logic.

Reply:
Also, would it be feasible to provide a way for pre-20295 users to trigger use of the system-default containerd? I don't know if it's really valuable, but it jumped out as a code-path that cannot be hit here.

Reply:
[Windows Server 2022 Preview (from the MS Evaluation Center)](https://cloudblogs.microsoft.com/windowsserver/2021/06/01/windows-server-2022-is-now-in-preview-on-the-evaluation-center/) is 20348, and I've seen people assuming that the RTM release will be a later patch on that build ID. However, there's no base images for the Windows containers for that release yet, the latest container base images are for the most-recent Windows Server Insider release (and probably last for this cycle), [20344](https://techcommunity.microsoft.com/t5/windows-server-insiders/announcing-windows-server-preview-build-20344/m-p/2303472).

I'm guessing [build 20295](https://techcommunity.microsoft.com/t5/windows-server-insiders/announcing-windows-server-preview-build-20295/m-p/2147144) was just the Windows Server Insider release at the time this was being worked on? Or was there some watermark moment that touched this?

AFAIK containerd works on RS5-Windows Server 2019 (their CI covers it, after all) but perhaps there's things Docker uses it for that are missing from the test coverage?

Reply:
Currently Win 2022 CI here on Moby run on build 20295 so those need to be updated if we want point to some other build. IIRC RS5 was first one which supports HCS v2 which containerd uses. Afaiu #42089 should add support for switching runtimes so question here is more likely that who we set defaults.

Reply:
After I left the above comment, I found #42527 which is now looking to upgrade the versions of the Windows Server 2022 Preview hosts.

That said, I think my wondering was "why 20295", in the sense of "20201 was also 'Windows Server 2022 LTSC' preview", so I was wondering if that build had any significance, e.g., OS fixes or features for container support landing.

If it was simply "That's the version on the CI cluster", that's good enough for now. I suspect all insider releases before 20348 will have expired before this ships anyway, so any version between 19042 and 20348 would be a reasonable watermark, as long as we note _why_ that version was picked.